js/single-docset-search.tsx

```diff
@@ -308,16 +349,57 @@ const search = (
   searchIndex: Index | undefined,
   query: string
 ): SearchResult[] | undefined => {
-  const results =
-    searchIndex &&
-    searchIndex.search(query, {
-      bool: "OR",
-      expand: true,
-      fields: {
-        name: { boost: 5 },
-        doc: { boost: 2 }
+  if (!searchIndex) {
+    return;
+  }
+
+  // we'd like to favour exact matches, ex: clojure.tools.test
+  const exactTokens = tokenize(query);
+  // but also entertain sub tokens, ex: clojure tools test
+  const subTokens = subTokenize(exactTokens);
+
+  const fieldQueries = [
+    { field: "name", boost: 10, tokens: exactTokens },
+    { field: "name", boost: 5, tokens: subTokens },
+    { field: "doc", boost: 4, tokens: exactTokens },
+    { field: "doc", boost: 2, tokens: subTokens }
+  ];
+
+  const queryResults: elasticlunr.SearchScores = {};
+
+  for (const fieldQuery of fieldQueries) {
+    const searchConfig = {
+      [fieldQuery.field]: { boost: fieldQuery.boost, bool: "OR", expand: true }
+    };
+
+    const fieldSearchResults: elasticlunr.SearchScores =
+      searchIndex.fieldSearch(
+        fieldQuery.tokens,
+        fieldQuery.field as keyof IndexItem,
+        searchConfig
+      );
+
+    for (const docRef in fieldSearchResults) {
+      fieldSearchResults[docRef] =
+        fieldSearchResults[docRef] * fieldQuery.boost;
```

Comment:
This could be:

```typescript
fieldSearchResults[docRef] *= fieldQuery.boost
```

Reply:
Aren't these already boosted because of `searchConfig` being passed in to the `searchIndex.fieldSearch`?

Reply:
Good question. Because the elasticlunr.js [search](https://github.com/weixsong/elasticlunr.js/blob/47100dab61cf6dc6184954cdf14ad55cc2540d4f/lib/index.js#L323-L361) did not support this kind of query I rolled my own, but also strongly based it on the existing search code [which does the extra boost](https://github.com/weixsong/elasticlunr.js/blob/47100dab61cf6dc6184954cdf14ad55cc2540d4f/lib/index.js#L342). Might be unnecessary, but conforms to current elasticlurn.js behaviour. So I think I'm ok with it.